### PR TITLE
feat: OKX Provider

### PR DESCRIPTION
**Pull request by davidterpay**

### CODE REVIEW

File: providers/websockets/okx/utils.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	// OKX provides a few different URLs for its Websocket API. The URLs can be found
+	// in the documentation here: https://www.okx.com/docs-v5/en/?shell#overview-production-trading-services
+	// The two production URLs are defined in ProductionURL and ProductionAWSURL. The
+	// DemoURL is used for testing purposes.
+
+	// ProductionURL is the public OKX Websocket URL.
+	ProductionURL = "wss://ws.okx.com:8443/ws/v5/public"
+
+	// ProductionAWSURL is the public OKX Websocket URL hosted on AWS.
+	ProductionAWSURL = "wss://wsaws.okx.com:8443/ws/v5/public"
```
Comment (davidterpay) [inline providers/websockets/okx/utils.go:13]: not used atm, but thought it would be useful to have here.

Comment (davidterpay) [reply providers/websockets/okx/utils.go:13]: we can maybe cycle the URLs when it restarts

File: providers/base/fetch.go [hunk 2]
```diff
@@ -101,15 +101,24 @@ func (p *BaseProvider[K, V]) attemptDataUpdate(ctx context.Context, responseCh c
...
+			if err := p.ws.Start(ctx, p.ids, responseCh); err != nil {
+				p.logger.Error("web socket query handler returned error", zap.Error(err))
+			}
 
-	if err := p.ws.Start(ctx, p.ids, responseCh); err != nil {
-		p.logger.Error("web socket query handler returned error", zap.Error(err))
-		return err
+			// If the web socket query handler returns, then the connection was closed. Wait for
+			// a bit before trying to reconnect.
+			time.Sleep(p.cfg.WebSocket.ReconnectionTimeout)
```
Comment (Eric-Warehime) [inline providers/base/fetch.go:119]: What happens if the remote client doesn't close the connection. I think the read will block indefinitely unless we [SetReadDeadline](https://pkg.go.dev/github.com/gorilla/websocket#Conn.SetReadDeadline).

Comment (davidterpay) [reply providers/base/fetch.go:119]: gonna add this as another thing we need to configure on the web socket providers.

Comment (davidterpay) [reply providers/base/fetch.go:119]: https://linear.app/skip/issue/BLO-758/figure-out-what-the-best-default-dial-configurations-are-for-web

**Review by Eric-Warehime — APPROVED** [review]

**Review by aljo242 — APPROVED** [review]

**Pull request merged**